engines/afalg/e_afalg.c

```diff
@@ -230,7 +230,11 @@ int afalg_fin_cipher_aio(afalg_aio *aio, int sfd, unsigned char *buf,
     memset(cb, '\0', sizeof(*cb));
     cb->aio_fildes = sfd;
     cb->aio_lio_opcode = IOCB_CMD_PREAD;
-    cb->aio_buf = (uint64_t)buf;
+    /*
+     * The pointer has to be converted to unsigned value first to avoid
+     * sign extension on cast to 64 bit value
+     */
+    cb->aio_buf = (uint64_t)(unsigned long)buf;
```

Comment:
Maybe something like the following would be helpful:

#if __x86_64__ && __ILP32__
cb->aio_buf = (uint64_t)(unsigned long)buf;
#else
cb->aio_buf = (uint64_t)buf;
#endif